### PR TITLE
add variants Notation and Action to Text component

### PR DESCRIPTION
**Pull request by feliciackh**
Blocked by #2571 
part of Theme 1.0 updates

### CODE REVIEW

**Review by feliciackh — COMMENTED** [review]
cs

File: packages/core/src/text/Text.tsx [hunk 1]
```diff
@@ -32,17 +28,17 @@ export type TextProps<T extends ElementType> = PolymorphicComponentPropWithRef<
...
      * Change text color palette
      */
     variant?: "primary" | "secondary";
   }
 >;
 
-type TextComponent = <T extends ElementType = "div">(
-  props: TextProps<T>
-) => ReactElement | null;
+type TextComponent = <T extends ElementType = "div">(props: TextProps<T>) => ReactElement | null;
```
Comment (joshwooding) [inline packages/core/src/text/Text.tsx:41]: What caused this formatting to be different?

**Comment by joshwooding** [comment]
This should probably go to main rather than theme-updates. We can merge this after the other PR

File: .changeset/little-pillows-fry.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+---
+"@salt-ds/core": minor
+---
+
+`Text` component has two new variants: `Action` and `Notation`. See [Storybook](https://storybook.saltdesignsystem.com/?path=/story/core-text-text-qa--all-variants-grid) for more information.
+
+```tsx
```
Comment (joshwooding) [inline .changeset/little-pillows-fry.md:7]: Are we avoiding mentioning Action here?

File: packages/core/src/text/Action.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+const withBaseName = makePrefixer("saltText");
+
+export const Action = forwardRef<
+  HTMLSpanElement,
+  Omit<TextProps<"span">, "as">
+>(function Action({ children, className, ...rest }, ref) {
+  return (
+    <Text
+      as="span"
+      className={clsx(className, withBaseName(`action`))}
```
Comment (joshwooding) [inline packages/core/src/text/Action.tsx:15]:
```suggestion
      className={clsx(className, withBaseName("action"))}
```

File: packages/core/src/text/Text.tsx [hunk 4]
```diff
@@ -32,7 +32,9 @@ export type TextProps<T extends ElementType> = PolymorphicComponentPropWithRef<
       | "label"
       | "display1"
       | "display2"
-      | "display3";
+      | "display3"
+      | "notation"
```
Comment (joshwooding) [inline packages/core/src/text/Text.tsx:36]: This won't work without the CSS being updated

File: .changeset/little-pillows-fry.md (hunk 2 above)
Comment (feliciackh) [reply .changeset/little-pillows-fry.md:7]: originally kept it succinct in theme PR but now that it's separated i've added Action too

File: packages/core/src/text/Display.tsx [hunk 5]
```diff
@@ -12,7 +12,7 @@ export const Display1 = forwardRef<
   return (
     <Text
       as="span"
-      className={clsx(className, withBaseName(`display1`))}
+      className={clsx(className, withBaseName("display1"))}
```
Comment (joshwooding) [inline packages/core/src/text/Display.tsx:15]: I know this was here already but isn't this equivalent to styleAs="display1" ?

File: .changeset/little-pillows-fry.md (hunk 2 above)
Comment (joshwooding) [reply .changeset/little-pillows-fry.md:7]: changesets should always cover the complete set of new additions

File: packages/core/src/text/Display.tsx (hunk 5 above)
Comment (feliciackh) [reply packages/core/src/text/Display.tsx:15]: yes they are, good spot i'll update these to use styleAs to encourage APi use

**Pull request closed without merging**